src/main/java/org/apache/accumulo/testing/performance/tests/GroupCommitPT.java

```diff
@@ -0,0 +1,234 @@
+
+      report.info("threadsTime" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(0))),
+          "Time it took the task to run in milliseconds");
+      report.info("threads" + i, i, "Number of threads");
+      report.info("batch" + i, 1, "Number of batches");
+      report.info("threadsMutations" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(1))),
+          "Total number of mutations");
+      report.result("threadsRate" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(2))),
+          "Total mutations per millisecond");
+    }
+
+    for (int i = 0; i < 6; i++) {
+      // This tests a single thread writing a different batch sizes of mutations,
+      // flushing after each batch. Group commit should approach these times for the same number
+      // mutations.
+      batchValues.clear();
+      batchValues = runBatch(env, report, 1, numThreads);
+
+      report.info("batchTime" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(0))),
+          "Time it took the task to run in milliseconds");
+      report.info("threads" + i, i, "Number of threads");
+      report.info("batch" + i, 1, "Number of batches");
+      report.info("batchMutations" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(1))),
+          "Total number of mutations");
+      report.result("batchRate" + i,
+          new Double(new DecimalFormat("#0.00").format(batchValues.get(2))),
+          "Total mutations per millisecond");
+    }
+
+    return report.build();
+  }
+
+  private ArrayList<Object> runBatch(Environment env, Report.Builder report, int numThreads,
+      int numToBatch) throws Exception {
+
+    String tableName = "mutslam";
+    env.getClient().tableOperations().create(tableName);
+
+    // scan just to wait for tablet be online
+    Scanner scanner = env.getClient().createScanner(tableName, Authorizations.EMPTY);
+    for (Map.Entry<Key,Value> entry : scanner) {
+      entry.getValue();
+    }
+
+    // number of batches each thread should write
+    int numToWrite = 100;
+
+    ArrayList<WriteTask> wasks = new ArrayList<WriteTask>();
+    ArrayList<Thread> threads = new ArrayList<Thread>();
+
+    for (int i = 0; i < numThreads; i++) {
+      WriteTask wask = new WriteTask(env.getClient().createBatchWriter(tableName,
+          new BatchWriterConfig().setMaxWriteThreads(1)), numToWrite, numToBatch);
+
+      wasks.add(wask);
+      Thread thread = new Thread(wask);
+      threads.add(thread);
+    }
+
+    for (Thread thread : threads) {
+      thread.start();
+    }
+
+    for (Thread thread : threads) {
+      thread.join();
+    }
+
+    long sum = 0;
+    for (WriteTask writeTask : wasks) {
+      sum += writeTask.getTime();
+    }
+
+    int totalNumMutations = numToWrite * numThreads * numToBatch;
+
+    ArrayList<Object> returnValues = new ArrayList<>();
```

Comment:
Would be a little bit cleaner to return a simple POJO rather than `ArrayList<Object>`